### PR TITLE
feat(tooltip,tourtip,infotip): added popper support

### LINKED ISSUE #1424: Infotip: overlay is obscured when placed inside of dialog
We've seen this occur a few times when infotips and tooltips are inside of elements with hidden overflow. 

Screen Shot 2021-04-29 at 12 05 55 PM

### PR DESCRIPTION
**Pull request by agliga**
Fixes #1424

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
* Added support for popperjs. Changed all out examples to use popper
* I removed the use of makeup expander, since all it seems like it was doing was to toggle `aria-expanded`. We can revisit this if needed, but it doesn't seem necessary.
* I did not touch any storybooks since they are static. The old code should work as if if the users opt not to use popper (so there is no breaking changes)

## Checklist
- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue
- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [X] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Comment by ianmcburnie** [comment]
Just noticed this on the popper documentation:

> Safari has unfortunate quirks regarding updating the position of DOM elements. We have filed WebKit bugs for these, so hopefully they will be fixed in the future.

Might be worth looking at those issues to make sure we are comfortable with them.

File: docs/_includes/infotip.html
Comment (ArtBlue) [inline docs/_includes/infotip.html:17]: Wouldn't this be a breaking change?

Comment (agliga) [reply docs/_includes/infotip.html:17]: Because this is only a docs change. The old code is still available for teams to use.

**Comment by agliga** [comment]
> > Just noticed this on the popper [documentation](https://popper.js.org/docs/v2/browser-support/#safari):
> 
> > 
> 
> > > Safari has unfortunate quirks regarding updating the position of DOM elements. We have filed WebKit bugs for these, so hopefully they will be fixed in the future.
> 
> > 
> 
> > Might be worth looking at those issues to make sure we are comfortable with them. We have to support back to Safari 12.
> 
> 
> 
> Those issues look to be related to scrolling, one of them over-scrolling (rubber band) effect. I can see why they would occur. If the intent of the library is to attach one element to another positionally and also to avoid viewport boundaries and reposition itself within the viewport, I can see how scrolling would create a "conflict of interest" here.

Yah from what i see that on safari it would act just like what we had before. So it wont be any different from our previous implementation of tip overlays.

File: docs/_includes/infotip.html
Comment (ArtBlue) [reply docs/_includes/infotip.html:17]: Oh, right. Cool.

**Review by ArtBlue — DISMISSED** [review]
LGTM.

File: docs/_includes/infotip.html
Comment (LuLaValva) [reply docs/_includes/infotip.html:17]: It won't be a breaking change for skin, but if we do this in eBayUI too we should tread lightly because there are a _lot_ of teams who override behavior of popovers to make them open in different directions.

Comment (agliga) [reply docs/_includes/infotip.html:17]: If they do that that's on them. They shouldn't be overriding base behavior 
But we do allow style overrides there. We probably should do something like
If they have a style override don't use popper to position.
If not then use popper.

Comment (ArtBlue) [reply docs/_includes/infotip.html:17]: I would prefer we not go down the road of checking for overrides and only using popper when none exist. I'd rather avoid various cases based on specific conditions if we can help it. If it comes down to it, I would actually prefer releasing this in a major version and warning people to remove overrides as part of the SWU.

Comment (agliga) [reply docs/_includes/infotip.html:17]: So I think there is a bit of misconception. First off this change would be only in ebayui. In skin there is no issue. In fact the old code works as is no problem (see storybook). 
This would only affect ebayui.

For ebayui I never said to not use popper if we have style overrides. We just don't use poppers automatic positioning. 
Popper allows you to override the position and setting offsets. What we would do in that case if they pass styles is to pass the position changes to popper and that would take care of it. 
So there wouldn't be anything breaking about it. It only only change internal implementation.

Comment (ArtBlue) [reply docs/_includes/infotip.html:17]: This is probably technically not a breaking change. The offsets are probably not an issue either; they serve as slight nudges. The positioning and overrides are the main thing I think.

Would we automatically send their styles into Popper without any checks? Or would we need to filter out the positioning ones and only send those to Popper? Would there be any sort of conditional overhead inside? If they override Popper's positioning, then the onus is on them to maintain it, right?

The main thing, I'd like to avoid is unnecessary positioning considerations being handled in ebay ui that later needs to be deprecated.

It's a bit hard conceptualizing this. Maybe if we had the ebay ui implementation, it might be easier to see if this poses any potential issue.

**Comment by ianmcburnie** [comment]
Is`[data-popper-placement]` added with JavaScript? Or is it declared in the base markup? I'm guessing the former as I don't see the attribute displayed in the example code. Which leads me to wonder whether we should be approaching this in a progressive enhancement fashion. Our existing modifiers and CSS work for most simple cases, so I don't think we need to always throw that away. It gives us a pretty good sensible default with the need for JS. Perhaps instead we should be progressively enhancing our existing modifiers (i.e. switching out `tooltip__pointer tooltip__pointer--left` for the popper equivalent). Let's chat offline about this.

**Comment by ianmcburnie** [comment]
Can we please add a test story for popper overlay inside of container with overflow hidden.

**Comment by ianmcburnie** [comment]
Not seeing the overlay or pointer appear in this example in Safari:

Screenshot 2023-10-03 at 12 56 35 PM

Here it is in Chrome:

Screenshot 2023-10-03 at 12 57 14 PM

**Comment by agliga** [comment]
Will wait for next release to get all the cleanup.

**Review by ArtBlue — APPROVED** [review]
Looks good!

**Comment by ianmcburnie** [comment]
On tourtips, the pointers look broken (in safari & chrome):
Screenshot 2023-12-18 at 11 26 42 AM
Screenshot 2023-12-18 at 11 26 36 AM

**Comment by ianmcburnie** [comment]
Isn't this Infotip pointer supposed to flip from bottom to top?

Screenshot 2023-12-18 at 11 29 25 AM


Screenshot 2023-12-18 at 11 29 36 AM

**Comment by agliga** [comment]
> Isn't this Infotip pointer supposed to flip from bottom to top when the info top repositions itself?
> 
> Screenshot 2023-12-18 at 11 29 25 AM Screenshot 2023-12-18 at 11 29 36 AM

Fixed, this was due to doing `!==` instead of `!=` as the docs recommended.

**Comment by agliga** [comment]
> On tourtips, the pointers look broken (in safari & chrome): Screenshot 2023-12-18 at 11 26 42 AM Screenshot 2023-12-18 at 11 26 36 AM

Added a 20px padding on arrow (The padding on the arrow is based on the border radius values so it doesn't look like this).

**Review by LuLaValva — APPROVED** [review]
Looks awesome! I'm excited to see this on other floating elements like `listbox` and `date-textbox`.

It would have probably been nice to do prettier for `main.js` in a separate PR but I'll let it slide

**Review by ianmcburnie — CHANGES_REQUESTED** [review]
I would like to still keep the examples of infotip, tooltip, tourtip that we had _without_ popper, and then added a dynamic positioning section for each which shows what accommodations we have to make in the markup/css (if any) for a 3rd party module like popper.

File: docs/_includes/tooltip.html
Comment (ianmcburnie) [inline docs/_includes/tooltip.html:7]:
```suggestion
```

File: docs/_includes/tooltip.html
Comment (ianmcburnie) [inline docs/_includes/tooltip.html:50]:
```suggestion
    <p>By default, the position of the tooltip and its pointer will remain static. For these following examples, we are using [floating-ui](https://floating-ui.com) to help position the tooltip and the pointer _dynamically_.</p>
```

File: docs/_includes/infotip.html
Comment (ianmcburnie) [inline docs/_includes/infotip.html:6]:
```suggestion
```

File: docs/_includes/infotip.html
Comment (ianmcburnie) [inline docs/_includes/infotip.html:61]:
```suggestion
    <p>By default, the position of the overlay and its pointer will remain static. For these following examples, we are using [floating-ui](https://floating-ui.com) to help position the tooltip and the pointer _dynamically_.</p>
```

File: docs/_includes/tourtip.html
Comment (ianmcburnie) [inline docs/_includes/tourtip.html:6]:
```suggestion
```

File: docs/_includes/tourtip.html
Comment (ianmcburnie) [inline docs/_includes/tourtip.html:53]:
```suggestion
    <p>By default, the position of the overlay and its pointer will remain static. For these following examples, we are using [floating-ui](https://floating-ui.com) to help position the tooltip and the pointer _dynamically_.</p>
```

**Comment by ianmcburnie** [comment]
Noticing a couple of discrepancies between positioning on prod storybook vs branch in tooltip stories:

Screenshot 2023-12-20 at 3 14 00 PM
Screenshot 2023-12-20 at 3 13 49 PM

Screenshot 2023-12-20 at 3 15 30 PM
Screenshot 2023-12-20 at 3 15 27 PM

**Review by ArtBlue — CHANGES_REQUESTED** [review]

File: docs/_includes/infotip.html
Comment (ianmcburnie) [reply docs/_includes/infotip.html:60]: Oops! That was my bad.

**Comment by agliga** [comment]
> Noticing a couple of discrepancies between positioning on prod storybook vs branch in tooltip stories:
> 
> tooltip -> expanded
> 
> Screenshot 2023-12-20 at 3 14 00 PM Screenshot 2023-12-20 at 3 13 49 PM
> Tooltip -> pointer -> left (and all others)
> 
> Screenshot 2023-12-20 at 3 15 30 PM Screenshot 2023-12-20 at 3 15 27 PM

Fixed this. I had removed the margin on bubble which was breaking in other places. I changed the positioning of the tip to get it to work.

File: docs/_includes/infotip.html
Comment (agliga) [reply docs/_includes/infotip.html:60]: Fixed this.

**Review by ArtBlue — COMMENTED** [review]
Looks fine to me, but we should get a confirmation approval from Ian since he had the bulk of feedback.

**Review by ianmcburnie — APPROVED** [review]
Great work @agliga

**Pull request merged**
